src/parser/jobs/drg/BuffDebuffUptime.js

```diff
@@ -0,0 +1,125 @@
+import React, {Fragment} from 'react'
+
+import {ActionLink} from 'components/ui/DbLink'
+import ACTIONS from 'data/ACTIONS'
+import STATUSES from 'data/STATUSES'
+import Module from 'parser/core/Module'
+import {Rule, Requirement} from 'parser/core/modules/Checklist'
+import {Suggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+
+
+//Playing it safe always. Gotta use protection
+const STATUS_DURATION = {
+	[STATUSES.PIERCING_RESISTANCE_DOWN.id]: 30000,
+	[STATUSES.CHAOS_THRUST.id]: 30000,
+
+}
+const HEAVY_THRUST_DURATION = 30000
+
+
+export default class Dots extends Module{
+	static handle = 'dots'
+	static dependencies=[
+		'checklist',
+		'combatants',
+		'cooldowns',
+		'enemies',
+		'invuln',
+		'suggestions',
+	]
+
+	_lastApplication = {}
+	_clip = {
+		[STATUSES.PIERCING_RESISTANCE_DOWN.id]: 0,
+		[STATUSES.CHAOS_THRUST.id]: 0,
+	}
+
+	constructor(...args){
+		super(...args)
+		const filter = {
+			by: 'player',
+			abilityId: [STATUSES.PIERCING_RESISTANCE_DOWN.id, STATUSES.CHAOS_THRUST.id],
+		}
+		this.addHook(['applydebuff', 'refreshdebuff'], filter, this._onDotApply)
+		this.addHook('complete', this._onComplete)
+	}
+
+	_onDotApply(event){
+		const statusId = event.ability.guid
+
+		const lastApplication = this._lastApplication[event.targetID] = this._lastApplication[event.targetID] || {}
+		if (!lastApplication[statusId]) {
+			lastApplication[statusId] = event.timestamp
+			return
+		}
+		//Base clip calc
+		let clip = STATUS_DURATION[statusId] - (event.timestamp - lastApplication[statusId])
+		// Removes untarget time from clipping
+		clip -= this.invuln.getUntargetableUptime('all', event.timestamp - STATUS_DURATION[statusId], event.timestamp)
+		// Capping the clip to 0
+		this._clip[statusId] += Math.max(0, clip)
+		lastApplication[statusId] = event.timestamp
+	}
+	_onComplete(){
+		//Checklist rules for uptimes
+		this.checklist.add(new Rule({
+			name: 'Keep up Disembowel & Chaos Thrust',
```

Comment:
Heavy thrust should either be a second checklist rule, or included in the title/description.

I'd almost say a second rule would make sense, it's not part of the same combo like the other two are.